### PR TITLE
Random pixel shift for dropped/thrown objects

### PR DESCRIPTION
**Pull request by Mickyan**
:cl:
add: Objects dropped and thrown will have a slight random pixel offset
/:cl:

Makes discarded objects (either dropped or thrown, manually or by explosions and such) look more organic, less likely to require alt clicking to see obscured objects when there's a lot of items on a single tile
Screenshot_1

### CODE REVIEW

**Comment by actioninja** [comment]
inb4 "now I can't precisely stack things from across the room, this is awful"

**Comment by SpaceManiac** [comment]
you can't precisely stack things from one tile away either

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -795,3 +799,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 /obj/item/proc/doStrip(mob/stripper, mob/owner)
 	return owner.dropItemToGround(src)
+
+/obj/item/proc/pixelOffset()
```
Comment (JJRcop) [inline code/game/objects/items.dm]: missing autodoc

**Comment by zxaber** [comment]
If we want the ability to stack things neatly, a compromise could be not having an offset when you throw something one tile.

It's not really like game critical, but it _is_ nice to be able to make a neat stack of metal.

**Comment by kriskog** [comment]
Eh, neat stacks are overrated anyway :+1:

**Comment by francinum** [comment]
If you want neat stacks use a rack.

**Comment by ike709** [comment]
The rack buff nobody knew we needed.

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -795,3 +799,8 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 /obj/item/proc/doStrip(mob/stripper, mob/owner)
 	return owner.dropItemToGround(src)
+
+///adds a random pixel offset, for non-precise item placement such as throwing or dropping to the ground
+/obj/item/proc/pixelOffset()
```
Comment (optimumtact) [inline code/game/objects/items.dm]: this should be a define imo

and also include random in the name

Comment (Mickyan) [reply code/game/objects/items.dm]: I thought about this but I'm not sure where to put it, is on top of this file okay?

Comment (SpaceManiac) [reply code/game/objects/items.dm]: I don't see why it would have to be a `#define`

Comment (optimumtact) [reply code/game/objects/items.dm]: @SpaceManiac it doesn't really make much sense as as a proc either, it's two lines

File: code/game/objects/structures/bedsheet_bin.dm [hunk 3]
```diff
@@ -34,6 +34,8 @@ LINEN BINS
 	if(layer == initial(layer))
 		layer = ABOVE_MOB_LAYER
 		to_chat(user, "<span class='notice'>You cover yourself with [src].</span>")
+		src.pixel_x = 0
```
Comment (ShizCalev) [inline code/game/objects/structures/bedsheet_bin.dm]: remove src

File: code/game/objects/structures/bedsheet_bin.dm [hunk 4]
```diff
@@ -34,6 +34,8 @@ LINEN BINS
 	if(layer == initial(layer))
 		layer = ABOVE_MOB_LAYER
 		to_chat(user, "<span class='notice'>You cover yourself with [src].</span>")
+		src.pixel_x = 0
+		src.pixel_y = 0
```
Comment (ShizCalev) [inline code/game/objects/structures/bedsheet_bin.dm]: .

File: code/game/objects/items.dm (hunk 2 above)
Comment (actioninja) [reply code/game/objects/items.dm]: just inline it.

**Comment by Mickyan** [comment]
I moved the proc call from dropped() to dropItemToGround(), this seems to make sense to me since the latter is called a lot less often but has nearly the same effect

One problem I've been struggling to fix with this is dumping storage items doesn't use this proc and I can't make heads or tails or how it even works

File: code/game/objects/items.dm (hunk 2 above)
Comment (Dennok) [reply code/game/objects/items.dm]: Why not set it to atom movable?

Comment (ninjanomnom) [reply code/game/objects/items.dm]: This really shouldn't be a proc or define at all

**Comment by SpaceManiac** [comment]
what color should we paint the random pixel offset shed

**Comment by ninjanomnom** [comment]
I like blue

**Comment by Mickyan** [comment]
Magenta

This is my shed so I get to pick

**Pull request merged**